pycparser/c_ast.py

```diff
@@ -55,7 +55,7 @@ def children(self):
         """
         pass
 
-    def show(self, buf=sys.stdout, offset=0, attrnames=False, nodenames=False, showcoord=False, _my_node_name=None):
+    def show(self, buf=sys.stdout, offset=0, attrnames=False, hideemptyattrs=False, nodenames=False, showcoord=False, _my_node_name=None):
```

Comment:
In general, I think it's preferable to have positive logic as much as possible, to avoid double-negation in code and reasoning about code.

Can you make it `showemptyattrs=True`?

Reply:
Hmm, yeah `show` does seem easier to reason about after extracting a helper.

```python
        if self.attr_names:
            is_empty = lambda v: v is None or (hasattr(v, '__len__') and len(v) == 0)
            nvlist = [(n, getattr(self,n)) for n in self.attr_names \
                      if showemptyattrs or not is_empty(getattr(self,n))]
```

Do you think it is worth also extracting a filter, or just leave it inline?

```python
        if self.attr_names:
            is_empty = lambda v: v is None or (hasattr(v, '__len__') and len(v) == 0)
            attr_filter = lambda v: showemptyattrs or not is_empty(v)
            nvlist = [(n, getattr(self,n)) for n in self.attr_names if attr_filter(getattr(self,n))]
```